plugin/trino-hive/src/main/java/io/trino/plugin/hive/HivePageSourceProvider.java

```diff
@@ -560,14 +562,14 @@ public static List<HiveColumnHandle> toColumnHandles(List<ColumnMapping> regular
                                     projectedColumn.getDereferenceIndices(),
                                     projectedColumn.getDereferenceNames(),
                                     fromHiveType,
-                                    fromHiveType.getType(typeManager));
+                                    fromHiveType.getType(typeManager, timestampPrecision));
```

Comment:
Am I understand correctly, that here was the main problem, that we lose precision, because of default method: 
```
    @Deprecated
    public TypeSignature getTypeSignature()
    {
        return getTypeSignature(DEFAULT_PRECISION);
    }
```

?

Reply:
Yes and we read all the timestamp inside a coercion column as `MILLISECOND` irrespective of the configuration.